workflow/controller/workflowpod.go

```diff
@@ -607,7 +609,7 @@ func (woc *wfOperationCtx) newExecContainer(name string, tmpl *wfv1.Template) *a
 }
 
 func isResourcesSpecified(ctr *apiv1.Container) bool {
-	return ctr != nil && (ctr.Resources.Limits.Cpu() != nil || ctr.Resources.Limits.Memory() != nil)
+	return ctr != nil && len(ctr.Resources.Limits) != 0
```

Comment:
Is there any reasoning I missed from the current logic? Please help double check this change here. At least the results from the new test cases are expected.